### DISCUSSION TITLE
Server sent events broadcasting is slow

### DISCUSSION THREAD

**Discussion by HoshenKadosh** [uNetworking/uWebSockets.js#806] [Q&A] [answered]
Hi everyone, with this library it's also possible to use it for sending SSE message, as in the example file for that.

It works great, until I wanted to broadcast a message to 16.5K clients. Therefore, I saved the connections in a map object and looped over them to write each http response the message.
However, this is very slow, max maybe 5 msg/sec (I assume because of the 'bridge' between nodejs and c++).

Is there anything to do about this? The websockets pub/sub works great with thousands of clients, it's very fast, it will be nice to have something like this for HTTP as well.

**Comment by uNetworkingAB** [comment, marked as answer]
If you do

```
res->cork(() => {
res->write(whatever);
}};
```

When calling from somewhere other than a callback from uWS you should see better perf.

**Reply by HoshenKadosh** [reply]
Thanks.
Is there any future plan to have pub/sub to HTTP responses as well in uwebsockets?

**Reply by HoshenKadosh** [reply]
@uNetworkingAB Any suggestions on how i can write and broadcast to many HTPP responses using the NodeJS wrapper?

**Comment by uNetworkingAB** [comment]
I think I get what you want but this is not something we support or have reason to support. Use websockets instead, they have everything you need and is an older standard so it's even more likely to be supported
